best-practices.md

```diff
@@ -623,3 +626,31 @@ basic geographic filtering from listeners.
 The dynamic STAC API would then listen to the notifications and update its internal datastore whenever new data comes into
 the static catalog. Implementors have had success using AWS Lambda to do a full 'serverless' updating of the elasticsearch
 database, but it could just as easily be a server-based process.
+
+## How to Differentiate STAC Files
+
+Any tool that crawls a STAC implementation or encounters a STAC file in the wild needs a clear way to determine if it is an Item, 
+Collection, Catalog or [ItemCollection](https://github.com/radiantearth/stac-api-spec/tree/v1.0.0-beta.1/fragments/itemcollection) 
+(part of the [STAC API spec](https://github.com/radiantearth/stac-api-spec/tree/v1.0.0-beta.1)). As of 1.0.0 this is done primarily
+with the `type` field, and secondarily in Items with `stac_version`, or optionally the `rel` of the link to it.
+
+```shell
+if type is 'Collection'
+  => Collection
+else if type is 'Catalog'
+  => Catalog
+else if type is 'Feature' and stac_version is defined
+  => Item
+else if type is 'FeatureCollection' and stac_version is defined
+  => ItemCollection
+else
+  => Invalid (JSON)
+```
+
+When actually crawling a STAC implementation one can also make use of the [relation type](catalog-spec/catalog-spec.md#relation-types
+) (`rel` field) when following a link. If it is an `item` rel type then the file must be a STAC Item. If it is `child`, `parent` or
+`root` then it must be a Catalog or a Collection, and the `type` field can be used to distinguish which one.
+
+In versions of STAC prior to 1.0 the process was a bit more complicated, as there was no `type` field for catalogs and collections.
+See [this issue comment](https://github.com/radiantearth/stac-spec/issues/889#issuecomment-684529444) for a heuristic that works
+for older STAC versions.
```

Comment:
maybe also add mention that Asset#type is the media type, rather than the GeoJSON/STAC type.

Reply:
Ok, added in some more language and links here, to help clarify.